### DISCUSSION TITLE
Correct way to configure jobs iOS

### DISCUSSION THREAD

**Discussion by Dipannita20** [shinyorg/shiny#908] [Q&A] [answered]
I have recently upgraded build environment and shiny to 2.4.3.3003
iOS 15.3
XCode 13.2.1
Visual Studio 2022
App has shiny jobs, notifications and locations nuget installed

Followed samples from shiny repo and updated my codebase for iOS as below

AppDelegate:
**Removed** 
`        public override void PerformFetch(UIApplication application, Action<UIBackgroundFetchResult> completionHandler) => this.ShinyPerformFetch(completionHandler);
`

info.plist:
```
<key>BGTaskSchedulerPermittedIdentifiers</key>
	<array>
		<string>com.shiny.job</string>
		<string>com.shiny.jobpower</string>
		<string>com.shiny.jobnet</string>
		<string>com.shiny.jobpowernet</string>
	</array>
```

Startup.cs:
```
services.UseJobs(true);
services.RegisterJob(new JobInfo(typeof(RefreshGeofenceRegionsJob))
            {
                RequiredInternetAccess = InternetAccess.Any,
                Repeat = true // this is the default
            });
```

If I trigger job manually for testing  it's a success
```
var result = await ShinyHost.Resolve<IJobManager>().RunAll(CancellationToken.None, false);
            return true;

```
But not seeing any notifications from the same job other time, it's been 2 hours now.

am I missing anything? Can someone please confirm the above setup.

I understand iPhone has it's own algorithm to run the job but how to test it out?

I build app on windows and with mac pair run on iPhone

Thanks in advanced

**Comment by aritchie** [comment, marked as answer]
https://samples.shinylib.net <= use them

This has also been answered here: https://github.com/shinyorg/shiny/discussions/888

**Comment by Dipannita20** [comment]
Thanks, job is working with <string>processing</string> enabled, some reason <string>fetch</string> is not working waited overnight.
I have gone through iOS documentation as well, not seeing much difference except time fetch is for 30 secs.


I setup job on Startup file, is there any configuration to delay the job by few minutes, not to run immediately? On Android job runs immediate as I open the app.
